src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -64,7 +64,12 @@ public TypeSymbol GetTypeInfo(SyntaxBase syntax)
             => GetTypeAssignment(syntax).Reference.Type;
 
         public IEnumerable<Diagnostic> GetAllDiagnostics()
-            => assignedTypes.Values.SelectMany(x => x.Diagnostics);
+        {
+            // ensure we've visited all of the syntax nodes
+            Visit(this.binder.FileSymbol.Syntax);

```

Comment:
>Visit(this.binder.FileSymbol.Syntax); [](start = 12, length = 37)

@shenglol would this remove the need for the manual type check on some of the missing nodes in your decorator error PR?

Reply:
Yeah it will. I'll update my PR once this is merged.